internal/it/test_skip.go

```diff
@@ -0,0 +1,221 @@
+				return true, err
+			}
+			skip := checkEnterprise(false)
+			if skip {
+				return true, nil
+			}
+		case parts[0] == "!"+skipEnterprise:
+			if err := validateLength(parts, 1, condition, "!enterprise"); err != nil {
+				return true, err
+			}
+			skip := checkEnterprise(true)
+			if skip {
+				return true, nil
+			}
+		case parts[0] == skipOSS:
+			if err := validateLength(parts, 1, condition, "oss"); err != nil {
+				return true, err
+			}
+			skip := checkEnterprise(true)
+			if skip {
+				return true, nil
+			}
+		case parts[0] == "!"+skipOSS:
+			if err := validateLength(parts, 1, condition, "!oss"); err != nil {
+				return true, err
+			}
+			skip := checkEnterprise(false)
+			if skip {
+				return true, nil
+			}
+		default:
+			return true, fmt.Errorf("Unexpected test skip constant \"%s\" in %s", parts[0], condition)
+		}
+	}
+	return false, nil
+}
+
+func validateLength(parts []string, expected int, condition, example string) error {
+	if len(parts) != expected {
+		return fmt.Errorf("Unexpected format for %s, example of expected condition: \"%s\" ", condition, example)
+	}
+	return nil
+}
+
+func checkVersion(given, operator, actual string) (bool, error) {
+	isGivenGreater, err := compareVersions(given, actual)
+	if err != nil {
+		return true, err
+	}
+
+	switch operator {
+	case "=":
+		if isGivenGreater == nil {
+			return true, nil
+		}
+		return false, nil
+	case "!=":
+		if isGivenGreater != nil {
+			return true, nil
+		}
+		return false, nil
+	case ">":
+		if isGivenGreater != nil && !*isGivenGreater {
+			return true, nil
+		}
+		return false, nil
+	case ">=":
+		if isGivenGreater != nil && *isGivenGreater {
+			return false, nil
+		}
+		return true, nil
+	case "<":
+		if isGivenGreater != nil && *isGivenGreater {
+			return true, nil
+		}
+		return false, nil
+	case "<=":
+		if isGivenGreater != nil && !*isGivenGreater {
+			return false, nil
+		}
+		return true, nil
+	default:
+		return true, fmt.Errorf("Unexpected test skip operator \"%s\" to compare versions", operator)
+	}
+}
+
+func compareVersions(given, actual string) (*bool, error) {
+	// versionNumbers describe the numbers of the present version
+	actualVersions := strings.Split(actual, ".")
+	// checkNumbers describe the numbers received to test for
+	givenVersions := strings.Split(given, ".")
+
+	min := len(givenVersions)
+	if min > len(actualVersions) {
+		min = len(actualVersions)
+	}
+
+	for i := 0; i < min; i++ {
+		givenNumber, err := strconv.Atoi(givenVersions[i])
+		if err != nil {
+			return nil, fmt.Errorf("Could not parse version number (to integer): %d", givenNumber)
+		}
+		actualNumber, err := strconv.Atoi(actualVersions[i])
+		if err != nil {
+			return nil, fmt.Errorf("Could not parse version number (to integer): %d", actualNumber)
+		}
+
+		if givenNumber > actualNumber {
+			res := true
+			return &res, nil
+		}
+		if actualNumber > givenNumber {
+			res := false
+			return &res, nil
+		}
```

Comment:
Although it's usually not a good idea, I think you can just do something like here:
```
num := givenNum - actualNum
if num != 0 {
    return num, nil
}
```

Reply:
I've cleaned up other parts of the code making it more compact. I think we can leave this here as it's a bit cleaner.